paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc

```diff
@@ -220,9 +220,14 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
      * ('any') which lets a primitive (convolution in this case) choose
      * the memory format preferred for best performance
      */
+    // TODO(jczaja): This is workaround to make grad op UT's numerical
+    // gradient computation proper as this op is called directly without
+    // fetch op following it , so numercial grad is computed (in python)
+    // using block formats which will give wrong results
     std::string data_format = ctx.Attr<std::string>("data_format");
     auto chosen_memory_format =
-        platform::data_format_to_memory_format(data_format);
+        is_test ? MKLDNNMemoryFormat::any
```

Comment:
is this correct?  should be like below ?

is_test ? platform::data_format_to_memory_format(data_format): MKLDNNMemoryFormat::any;

Reply:
yes. Till GRAD NHWC is not working this is to be like that. Do you have any question on that one?

Reply:
Yes. When there is inference/test then mkl-dnn is free to choose conv implementation (MKLDNNMemoryFormat::any) . When there is a training we check if user enforced *data_format* and if positive then there are two use cases: AnyLayout and NHWC/NCHW .  For AnyLayout again MKL-DNN is free to choose implementation of conv for NCHW/NHWC MKL-DNN will use NCHW/NHWC implementation. This is needed for GRAD Unit tests to pass. As mentioned earlier using data_format to enforce implementation of MKL-DNN is leading to low performance , but this already happened in previous PRs , this PR does not make any additional harm and make sure performance is good for inference.